account_chart_update/wizard/wizard_chart_update.py

```diff
@@ -337,18 +393,61 @@ def padded_code(self, code):
     @tools.ormcache("templates")
     def find_account_by_templates(self, templates):
         """Find an account that matches the template."""
-        return self.env['account.account'].search(
-            [('code', 'in', map(self.padded_code, templates.mapped("code"))),
-             ('company_id', '=', self.company_id.id)],
-        ).id
+        account_model = self.env['account.account']
+        for matching in self.account_matching_ids.sorted('sequence'):
+            if matching.matching_value == 'xml_id':
+                data = self.env['ir.model.data'].search([
+                    ('name', 'in', map(self._get_real_xml_name, templates))
+                ])
+                if not data:
+                    continue
+                criteria = ('id', 'in', data.mapped('res_id'))
+            elif matching.matching_value == 'code':
+                codes = templates.mapped("code")
+                if not codes:
+                    continue
+                criteria = ('code', 'in', map(self.padded_code, codes))
+            else:
+                field_name = matching.matching_value
+                field_values = templates.mapped(field_name)
+                if not field_values:
+                    continue
+                criteria = (field_name, 'in', field_values)
+
+            result = account_model.search(
+                [criteria, ('company_id', '=', self.company_id.id)])
+            if result:
+                return result.id
+
+        return False
 
     @api.multi
     @tools.ormcache("templates")
     def find_fp_by_templates(self, templates):
         """Find a real fiscal position from a template."""
-        return self.env['account.fiscal.position'].search(
-            [('name', 'in', templates.mapped("name")),
-             ('company_id', '=', self.company_id.id)], limit=1).id
+        fp_model = self.env['account.fiscal.position']
+        for matching in self.fp_matching_ids.sorted('sequence'):
+            if matching.matching_value == 'xml_id':
```

Comment:
Why not using `self.env.ref(xml_id, False)` instead?

Reply:
done